example_test.go

```diff
@@ -10,6 +10,131 @@ import (
 	"inet.af/netaddr"
 )
 
+func ExampleIP() {
+	ip, err := netaddr.ParseIP("192.0.2.3")
+	if err != nil {
+		panic(err)
+	}
+
+	// netaddr.IP supports comparison using ==
+	fmt.Println(ip == netaddr.IPv4(192, 0, 2, 3))
+
+	// netaddr.IP can be used as a map key
+	hosts := make(map[netaddr.IP]string)
+	hosts[ip] = "example.net"
+	fmt.Println(hosts)
+	// Output:
+	// true
+	// map[192.0.2.3:example.net]
+}
+
+func ExampleIP_Is4() {
+	ipv4 := netaddr.MustParseIP("192.0.2.3")
+	ipv6 := netaddr.MustParseIP("2001:db8::68")
+	ipEmpty := netaddr.IP{}
```

Comment:
Suggest

```go
var zeroIP netaddr.IP
```

and throughout.

Reply:
Can do. Do you think changing `ipv4`, and `ipv6` to `v4IP` and v6IP` would be a good matching change?

Alternatively eliminating the variables entirely has some appeal.

Reply:
Alternatively `var nilIP IP`, as used in

https://github.com/inetaf/netaddr/blob/d57edf19c51753b7d583a5f5054d7d1d2d3cb1ad/netaddr_test.go#L552-L554

Reply:
I'd keep ipv4 and ipv6 as-is. I'm ok with zero or zeroIP, or even ipZero. I'd rather not use nilIP.